Add jsdoc for `debug` and `introspection` in `GraphQLModule`

### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

No

### Describe the solution you'd like

Add jsdoc for the keys `debug` and `introspection`

### Teachability, documentation, adoption, migration strategy

Auto-completion along with jsDocs can help users understand the use case of each.

### What is the motivation / use case for changing the behavior?

Currently, the keys seem to not exist in the interface and hence, autocomplete does not work.

@kamilmysliwiec I'm getting npm errors

```log
C:\Users\aaron\Documents\projects\nestjs-graphql>npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: @nestjs/graphql-workspace@0.0.0
npm ERR! Found: prettier@2.8.8
npm ERR! node_modules/prettier
npm ERR!   dev prettier@"2.8.8" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer prettier@">=3.0.0" from eslint-plugin-prettier@5.1.3
npm ERR! node_modules/eslint-plugin-prettier
npm ERR!   dev eslint-plugin-prettier@"5.1.3" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\aaron\AppData\Local\npm-cache\_logs\2024-02-12T18_51_19_689Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: C:\Users\aaron\AppData\Local\npm-cache\_logs\2024-02-12T18_51_19_689Z-debug-0.log

```

Node version: 18.19.0
npm version: 10.2.3

--- a/packages/graphql/lib/interfaces/gql-module-options.interface.ts
+++ b/packages/graphql/lib/interfaces/gql-module-options.interface.ts
@@ -120,6 +120,20 @@ export interface GqlModuleOptions<TDriver extends GraphQLDriver = any> {
    * Extra static metadata to be loaded into the specification
    */
   metadata?: () => Promise<Record<string, any>>;
+
+  /**
+   * If `true`, enables development mode helpers and logs messages of all severity levels
+   * If `false`, only warn- and error-level messages are logged.
+   * @default true
+   */
+  debug?: boolean;
+
+  /**
+   * If `true`, enables schema introspection by clients. Default is `true` unless `NODE_ENV` is
+   * set to `production`
+   * @default true
+   */
+  introspection?: boolean;
 }
 
 export interface GqlOptionsFactory<

--- a/packages/graphql/lib/schema-builder/metadata/union.metadata.ts
+++ b/packages/graphql/lib/schema-builder/metadata/union.metadata.ts
@@ -1,7 +1,9 @@
 import { Type } from '@nestjs/common';
 import { ResolveTypeFn } from '../../interfaces';
 
-export interface UnionMetadata<T extends readonly Type<unknown>[] = readonly Type<unknown>[]> {
+export interface UnionMetadata<
+  T extends readonly Type<unknown>[] = readonly Type<unknown>[],
+> {
   name: string;
   typesFn: () => T;
   id?: symbol;

--- a/packages/graphql/lib/type-factories/create-union-type.factory.ts
+++ b/packages/graphql/lib/type-factories/create-union-type.factory.ts
@@ -13,7 +13,9 @@ import { TypeMetadataStorage } from '../schema-builder/storages/type-metadata.st
 /**
  * Interface defining options that can be passed to `createUnionType` function.
  */
-export interface UnionOptions<T extends readonly Type<unknown>[] = Type<unknown>[]> {
+export interface UnionOptions<
+  T extends readonly Type<unknown>[] = Type<unknown>[],
+> {
   /**
    * Name of the union.
    */
@@ -40,9 +42,9 @@ export type Union<T extends readonly any[]> = InstanceType<ArrayElement<T>>;
  * Creates a GraphQL union type composed of types references.
  * @param options
  */
-export function createUnionType<T extends readonly Type<unknown>[] = Type<unknown>[]>(
-  options: UnionOptions<T>,
-): Union<T> {
+export function createUnionType<
+  T extends readonly Type<unknown>[] = Type<unknown>[],
+>(options: UnionOptions<T>): Union<T> {
   const { name, description, types, resolveType } = options;
   const id = Symbol(name);
 

